src/poetry/mixology/version_solver.py

```diff
@@ -374,10 +375,20 @@ def _get_min(dependency: Dependency) -> Tuple[bool, int]:
                 )
                 return dependency.complete_name
 
-            try:
-                version = packages[0]
-            except IndexError:
-                version = None
+            version = None
+            if dependency.name not in self._use_latest:
+                # prefer locked version of compatible (not exact same) dependency;
+                # required in order to not unnecessarily update dependencies with
+                # extras, e.g. "coverage" vs. "coverage[toml]"
+                locked = self._locked.get(dependency.name, None)
+                if locked is not None:
```

Comment:
There's not a good way to unroll the nested logic here, is there? This is a bit dense, but I'm not seeing an obvious simplification myself.

Reply:
If you want to refrain from taking advantage of the fact that the list is sorted and thus from the early return if the version is not found, you can simplify the loop as follows:

```
if locked is not None:
    for version in packages:
        if version.version == locked.version:
            break
```

Reply:
In that case I'm wondering if `next()` wouldn't be clearer -- e.g.

```python
    version = next((p.version for p in packages if p.version == locked.version), None)
```

Thoughts?

Reply:
I like it, but it has to be:

`version = next((p for p in packages if p.version == locked.version), None)`

Maybe a bit confusing that the variable `version` is a `package`. Should we rename it to `package`?

Reply:
@neersighted Sorry for bothering, but I had the impression that the review of this PR has already been well advanced - maybe, the naming of the variable being the only point to be clarified before merging. (Correct me if I'm wrong.) I'd appreciate if you find some minutes so that we may finalize it.